### PR TITLE
feat: add video title and an endpont to edit the video title

### PR DESCRIPTION
**Pull request by DestinedCodes**
## Description
This pull request introduces a new feature to the application by adding the ability to assign a title to videos. The title will enhance the user experience by providing a descriptive identifier for each video.

## Changes Made
- Added a 'title' field to the Video model in `app/models/video_models.py`.
- Modified the `start_recording` function in `app/routes/video_routes.py` to generate a default title for new videos.
- Implemented an endpoint to update the title of a video.
- Updated the endpoint to get a specific video to use the video ID instead of username and video ID.

## Related Issue
No specific issue is related to this feature.

## Motivation and Context
The motivation behind this feature is to provide users with the ability to name their videos, making it easier to identify and manage their content.

## How Has This Been Tested?
- Manually tested by creating a video and updating its title.
- Unit tests were added to test the new functionality.

## Screenshot
Screenshot_20231009-172219
Screenshot_20231009-172307
Screenshot_20231009-172335

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/routes/video_routes.py [hunk 1]
```diff
@@ -59,9 +59,11 @@ def start_recording(
         db.commit()
         db.refresh(new_user)
 
+    video_id=generate_id()
```
Comment (zxenonx) [inline app/routes/video_routes.py:62]:
```suggestion
    video_id = generate_id()
```

File: app/routes/video_routes.py [hunk 2]
```diff
@@ -294,6 +296,34 @@ def get_thumbnail(video_id: str, db: Session = Depends(get_db)):
...
+    """
+    if video := db.query(Video).filter(Video.id == video_id).first():
+        video.title = title
+        db.commit()
+        db.close()
+
+        return {"msg": "Title updated successfully!"}
+
+    db.close()
+    raise HTTPException(status_code=404, detail="Video not found.")
```
Comment (zxenonx) [inline app/routes/video_routes.py:326]:
```suggestion
    video = db.query(Video).filter(Video.id == video_id).first()
    if not video:
       raise HTTPException(status_code=404, detail="Video not found.")
     video.title = title
     db.commit()
     db.close()
    return {"msg": "Title updated successfully!"}

    db.close()
    
```

File: app/routes/video_routes.py [hunk 3]
```diff
@@ -294,6 +296,34 @@ def get_thumbnail(video_id: str, db: Session = Depends(get_db)):
         raise HTTPException(status_code=404, detail="Video not found.")
     return FileResponse(video.thumbnail_location, media_type="image/jpeg")
 
+@router.put("/video/{video_id}")
```
Comment (zxenonx) [inline app/routes/video_routes.py:299]: 'Put' obviously works but let's use Patch here as it's just a partial update

**Review by zxenonx — CHANGES_REQUESTED** [review]
Good job @DestinedCodes 
I left some comments

File: app/routes/video_routes.py (hunk 3 above)
Comment (DestinedCodes) [reply app/routes/video_routes.py:299]: Thanks so much. Since I'm updating just that field I'll use PUT

**Review by zxenonx — APPROVED** [review]
Good

**Pull request merged**